src/controllers/legacycontrollersettings.h

```diff
@@ -0,0 +1,124 @@
+#pragma once
+
+#include <QDomElement>
+#include <QJSValue>
+#include <QLabel>
+#include <QLayout>
+#include <QList>
+#include <QObject>
+#include <QSizePolicy>
+#include <QSpinBox>
+#include <QString>
+#include <QWidget>
+#include <limits>
+#include <memory>
+#include <tuple>
+
+#include "controllers/legacycontrollersettingsfactory.h"
+
+/// @brief The abstract controller setting. Any type of setting will have to
+/// implement this base class
+class AbstractLegacyControllerSetting : public QObject {
+    Q_OBJECT
+  public:
+    virtual ~AbstractLegacyControllerSetting() = default;
```

Comment:
```suggestion
    ~AbstractLegacyControllerSetting() override = default;
```
We use override whenever we inherit a virtual function. This makes sure that the base class function is still virtual after refactoring,